core/src/main/java/org/apache/calcite/plan/volcano/VolcanoPlanner.java

```diff
@@ -982,11 +982,17 @@ void propagateCostImprovements(RelNode rel) {
         if (!relNode.getTraitSet().satisfies(subset.getTraitSet())) {
           continue;
         }
-        if (!cost.isLt(subset.bestCost)) {
+
+        // Update subset best and best's cost when we find a cheaper rel
```

Comment:
these two comments do not really seem to describe what the code next to them is doing

Reply:
This is the condition when we need to update the best and best's cost.

Reply:
The line commented does not update anything, it just calls continue.
The comment should say the opposite: if the cost is higher, continue.